hack/local/README.md

```diff
@@ -108,3 +109,91 @@ To change or add packages, see the Jupyter Dockerfile ([**Dockerfile.jupy**](doc
+The MLRun API service needs to have permissions to manipulate Vault k8s roles, Vault policies and of course handle secrets. The Vault authentication
+method within MLRun uses the following logic:
+1. If a Vault Token is available in the `MLRUN_VAULT_TOKEN` environment variable, use it to authenticate directly with Vault (not through k8s authentication). For example, if you
+have the root token available and wants to use it, you can set it directly in the API pod env variables
+2. Otherwise, a JWT token is used to authenticate using k8s mode. The JWT token can be in one of two places:
+
+    a.  As part of the default service-account for the pod, in the standard path of `/var/run/secrets/kubernetes.io/serviceaccount/token`. If this is your
+    chosen approach, then other than setting the API pod's service account properly, nothing else is needed.
+    
+    b. In a path specified in MLRun conf under `vault.token_path` - the default for that is `~/.mlrun/vault`, but can be overridden like any other MLRun configuration -
+    to override set the `MLRUN_VAULT__TOKEN_PATH` to the place where you want to mount your token. This approach is handy if you want to utilize a different SA than the 
+    pod's default SA to authenticate with Vault. In that case, simply extract the token from the SA that you want to use and mount it in the location pointed at by the variable.
+     
+#### Working with k8s authentication and a JWT token
+If a JWT token was used (either of the approaches in 2a above), a proper Vault role must be provided. The role, together with the SA, are used by Vault to determine the Vault policy granted to the API.
+The role is specified through the `MLRUN_VAULT_ROLE` environment variable, and can be in one of two forms:
+1. `project:<project name>` - specifies a project-level context. 
+    Usually you would not use that for the MLRun API pod, as it needs access to secrets of multiple projects.
+2. `user:<user name>` - specifies a user-level context.
+
+To configure the MLRun API, the recommended approach is:
+1. Set the MLRun Vault URL using the `MLRUN_VAULT___URL` environment variable. 
+    For exmaple, If using Minikube on Mac it should be set to `http://docker.for.mac.localhost:8200`
+2. Set the Vault role for the pod to be `user:admin`
+3. Make sure the service-account set for the MLRun API pod is `mlrun-api`
+4. Configure a Vault role using the following command:
+
+    `$ vault write auth/kubernetes/role/mlrun-role-user-admin bound_service_account_names=mlrun-api bound_service_account_namespaces=default-tenant policies=admin_full ttl=12h`
+
+5. Create the `admin_full` Vault policy, with the following content:
+    ```yaml
+    # Allow access to admin user-context secrets
+    path "secret/data/mlrun/users/admin" {
+      capabilities = ["read", "list", "create", "update", "sudo"]
+    }
+    
+    path "secret/data/mlrun/users/admin/*" {
+      capabilities = ["read", "list", "create", "update", "sudo"]
+    }
```

Comment:
Actually, the 2nd one is not needed **for now** since all secrets are added as key-values in the `secret/data/mlrun/users/admin` path. The 2nd is more of a hachana lemazgan to keep it ready for sub-directories and keys within the user namespace.
I prefer to keep both rather than deal with adding the 2nd one later and handling upgrade flows - the policy as-is is not very complicated and it's all being done in the backend anyway without burdening the user.